spec/models/validators/valid_date_presence_validator_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe ValidDatePresenceValidator do
+  let(:test_model) do
+    Class.new do
+      include ActiveModel::Validations
+
+      attr_accessor :date_attr, :other_attr
+
+      validates :date_attr, valid_date_presence: true
+    end
+  end
+
+  subject { test_model.new }
+
+  it 'passes when the date is valid' do
+    expect_any_instance_of(ValidDatePresenceValidator).to receive(:valid_date?).and_return true
```

Comment:
You are right. I forgot that this validator outsources `valid_date?` to a module.